### ISSUE TITLE
[feat] add `react-x/no-nested-lazy-component-declarations`

### ISSUE THREAD

**Issue by Rel1cx** [Rel1cx/eslint-react#1048]
## Lazy component’s state gets reset unexpectedly

Do not declare lazy components inside other components:

```tsx
import { lazy } from 'react';

function Editor() {
  // 🔴 Bad: This will cause all state to be reset on re-renders
  const MarkdownPreview = lazy(() => import('./MarkdownPreview.js'));
  // ...
}
```

Instead, always declare them at the top level of your module:

```tsx
import { lazy } from 'react';

// ✅ Good: Declare lazy components outside of your components
const MarkdownPreview = lazy(() => import('./MarkdownPreview.js'));

function Editor() {
  // ...
}
```

### Describe the solution you'd like

Add a `react-x/no-nested-lazy-component-declarations` rule to check it.

### Alternatives considered

Enhance the `react-x/no-nested-component-definitions` rule to check lazy component declarations.
